stdlib/public/core/ManagedBuffer.swift

```diff
@@ -136,6 +136,26 @@ open class ManagedBuffer<Header, Element> {
   }
 }
 
+@inline(never)
+public func reallocateUniquelyReferenced<Header, Element, Buffer: ManagedBuffer<Header, Element>>(
+  buffer: inout Buffer,
+  newMinimumCapacity: Int
+) {
+  precondition(_isBitwiseTakable(Header.self))
+  precondition(_isBitwiseTakable(Element.self))
+  precondition(isKnownUniquelyReferenced(&buffer))
+  precondition(!_hasSideTable(&buffer))
```

Comment:
Is this the only place where hasSideTable is used?

It feels that adding a built-in + instruction for this is the wrong approach. Why not check this in the _swift_stdlib_realloc function?

Reply:
Also, AFAIK side tables are also created for ref count overflows. So you cannot be sure if the precondition will hold, even if there are no weak references.

Reply:
Took me a while to page this info back in, but the only refcount that overflows currently is the unowned count. If an object has never had weak or unowned references (which I think we can reasonably require for this use case) then it won't have a side table. At least as things stand now.

Reply:
@weissi Any assumptions that we make about this use case at least need to be documented as part of the ManagedBuffer contract. e.g. weak/unowned references are disallowed. Otherwise, we can just get rid of the precondition and call the slow path for _hasSideTable.

@mikeash How are associated objects implemented? Don't they require some kind of side table as well?

Reply:
Relocating an object like this will break associated objects in interesting ways. That's another thing we should be OK to forbid here. I'm not sure if it's practical to assert it, though.

If Swift ever gets something like associated objects that doesn't go through the ObjC runtime, it will likely use the side table mechanism, so that would be another potential source of them in the future.